DDEyC-Assistant/Services/ChatService.cs

```diff
@@ -442,22 +445,55 @@ private async Task<string> HandleGetJobListingsAsync(string arguments)
 
         try
         {
-            var queryParams = BuildJobListingQueryParams(args);
-            var url = $"{_configuration["AppSettings:BackEndUrl"]}/api/joblistings";
+            var queryParams = new List<KeyValuePair<string, string>>();
 
-            // QueryHelpers.AddQueryString accepts IEnumerable<KeyValuePair<string, string>> directly
-            var response = await _httpClient.GetAsync(QueryHelpers.AddQueryString(url, queryParams));
-            if (response.IsSuccessStatusCode)
+            if (!string.IsNullOrEmpty(args.Query))
```

Comment:
use switch based in c# 8

Reply:
https://www.c-sharpcorner.com/article/c-sharp-8-0-new-feature-swtich-expression/

Reply:
This is a progressive query builder, it is built on sequential ifs, a switch statement is not adequate for this function, no argument is compared twice or against more than the two values, null or empty, the conditions are not mutually exclusive and no two conditions do the same thing